### PR TITLE
Fix #775: Introduce SASS_BINARY_SITE environment, refactor tests/api.js

### LINKED ISSUE #775: Impossible to mirror the libsass binaries
After having investigated the proxy issue, I found out an extra problem: Even when the proxy call is made to work, my corporate proxy does not allow downloading native executables (.so, .dll, etc..) from the internet. The binaries provided by node-sass are blocked by the proxy and prevent the proper installation of node-sass.

As a workaround for this kind of issues, my organisation has an internal server dedicated to hosting trusted binaries that can be accessed without going through the proxy. That sounded like a reasonable solution to my problem until I disovered how the path to the binary is resolved:

```javascript
function getBinaryUrl() {
  return flags.binaryUrl ||
         package.nodeSassConfig ? package.nodeSassConfig.binaryUrl : null ||
         process.env.SASS_BINARY_URL ||
         ['https://github.com/sass/node-sass/releases/download//v',
          `package.version, '/', sass.binaryName].join('');
}
```
and 
```javascript
function getBinaryName() {
  var binaryName;

  if (flags.binaryName) {
    binaryName = flags.binaryName;
  } else if (package.nodeSassConfig && package.nodeSassConfig.binaryName) {
    binaryName = package.nodeSassConfig.binaryName;
  } else if (process.env.SASS_BINARY_NAME) {
    binaryName = process.env.SASS_BINARY_NAME;
  } else {
    binaryName = [process.platform, '-',
                  process.arch, '-',
                  process.versions.modules].join('');
  }

  return [binaryName, 'binding.node'].join('_');
}
```

This means that if I want to override the location of the binary, I need to point to the exact location of the particular binary that works for each machine's platform/architecture/module version/node-sass version combination. This is currently pretty hard to do given the various environments that our build scripts must run on.

node-sass should provide an option, say `--sass-binary-mirror-url`, that allows me to build a dumb mirror of all the binaries and that tells the node-sass installer to perform its binary resolution algorithm using this URL as base (instead of the hardcoded `https://github.com/sass/node-sass/releases/download/`)

As a workaround, I guess I could copy both of these methods to my `gulpfile.js` and compute the binary URL there. However, I'm not sure how I could make node-sass pick up that url. Is it possible?

### PR DESCRIPTION
**Pull request by saper**
Factor out `tests/render.js` out of `tests/api.js`,
reducing boilerplate and allowing easier save and
restore of the environment.

Add tests for #775

### CODE REVIEW

**Comment by am11** [comment]
The test for #775 should go in #775. You can combine the refactoring of test in one PR. Would be easy to review, monitor CI progress and since so forth.

**Comment by saper** [comment]
This pull request also includes #847; I propose we discuss refactoring of tests here - it makes no sense to introduce another PR just for commit 565ece9 and commit 31321b9 serves as a justification why I needed to refactor this. Unfortunately, unlike in Gerrit Code Review, I cannot stack pull requests on top of another. If you think that `testes/render.js` is unproblematic, we can merge this and #847 closes. If not, let's discuss testing here and push #847 alone, as it's trivial.

**Comment by xzyfer** [comment]
I'll look at this post #775.

File: lib/extensions.js [hunk 1]
```diff
@@ -76,21 +76,40 @@ function getBinaryName() {
...
-         package.nodeSassConfig ? package.nodeSassConfig.binaryUrl : null ||
-         process.env.SASS_BINARY_URL ||
-         ['https://github.com/sass/node-sass/releases/download/v',
-          package.version, '/', sass.binaryName].join('');
+  var site = flags['--sass-binary-site'] || process.env.SASS_BINARY_SITE  ||
+             (package.nodeSassConfig ? package.nodeSassConfig.binarySite : null);
+  if (site) {
+		  return [site, 'v' + package.version, sass.binaryName].join('/');
+  } else {
+          return null;
```
Comment (xzyfer) [inline lib/extensions.js]: This shoyldn't be possible. We should default to `'https://github.com/sass/node-sass/releases/download/v` if the users has give no proeference.

File: package.json [hunk 2]
```diff
@@ -18,6 +18,9 @@
     "node": ">=0.10.0"
   },
   "main": "lib/index.js",
+  "nodeSassConfig": {
+    "binarySite1": "https://github.com/sass/node-sass/releases/download"
```
Comment (xzyfer) [inline package.json]: typo? `binarySite1`

**Comment by xzyfer** [comment]
Superseded by https://github.com/sass/node-sass/pull/901.

**Pull request closed without merging**